include/votca/xtp/bse.h

```diff
@@ -56,31 +42,34 @@ struct Population {
         _bse_singlet_coefficients_AR(orbitals.BSESingletCoefficientsAR()),
         _bse_triplet_energies(orbitals.BSETripletEnergies()),
         _bse_triplet_coefficients(orbitals.BSETripletCoefficients()),
-        _min_print_weight(min_print_weight){};
-  
-  void setGWData(const TCMatrix_gwbse* Mmn,const PPM* ppm,const Eigen::MatrixXd* Hqp){
-      _Mmn=Mmn;
-      _ppm=ppm;
-      _Hqp=Hqp;   
-  }
+        _Mmn(Mmn),_Hqp(Hqp){};
+
+    struct options{
+        bool useTDA=true;
+        int homo;
+        int rpamin;
+        int rpamax;
+        int qpmin;
+        int vmin;
+        int cmax;
+        int nmax; //number of eigenvectors to calculate
+        double min_print_weight=0.5;  //minimium contribution for state to print it
+        };
   
-  void setBSEindices(int homo,int vmin, int cmax, int nmax) {
-                _homo=homo;
-                _bse_vmin = vmin;
-                _bse_vmax = homo;
-                _bse_cmin = homo+1;
-                _bse_cmax = cmax;
-                _bse_nmax = nmax;
-                _bse_vtotal = _bse_vmax - _bse_vmin + 1;
-                _bse_ctotal = _bse_cmax - _bse_cmin + 1;
-                _bse_size = _bse_vtotal * _bse_ctotal;
-                return;
-            }
+ 
 
-   
-  void Solve_triplets();
+   void configure(const options& opt){
```

Comment:
I think that the use `options` make the code more robust by isolating encapsulating the options in a single entity